controllers/gitrepository_controller.go

```diff
@@ -117,14 +116,18 @@ type GitRepositoryReconciler struct {
 	Storage        *Storage
 	ControllerName string
 
-	requeueDependency time.Duration
-	features          map[string]bool
+	libgit2TransportInitialized func() bool
+	requeueDependency           time.Duration
+	features                    map[string]bool
 }
 
 type GitRepositoryReconcilerOptions struct {
 	MaxConcurrentReconciles   int
 	DependencyRequeueInterval time.Duration
 	RateLimiter               ratelimiter.RateLimiter
+	// Libgit2TransportInitialized lets the reconciler know whether
+	// libgit2 transport was intialized successfully.
+	Libgit2TransportInitialized func() bool
```

Comment:
I don't this this is an option. If it's not set, the reconciler will panic. Similar to `HelmRepositoryOCIReconciler`'s `RegistryClientGenerator`, it should be a necessary field.